0.13 announcement post

alice-i-cecile opened the pull request:
Since @cart is drowning in admin paperwork for the legal org already (thanks!), I'm going to lead the effort here this cycle.

Like in #754, I'll stub out the major sections here, and then folks can tackle and refine them in their own PRs to this branch :)

TODOs will be tracked in the post, with the handy `TODO` marker.

nicopap commented:
Following is a list of potential sections for the news post. I browsed the commits since November 4th.

**If you think other sections should be included, please say it**. This list is not exhaustive.

Between parenthesis **potential section authors**. `?@author` means "I think you could write the section", but doesn't mean it is claimed. I've put myself where I'd be glad to claim the section.

- `winit` and `wgpu` updates
  - [ ] `winit` 0.29 (?@Vrixyz, @nicopap)
  - [ ] `wgpu` 0.19 (?@elabajaba)
- [ ] Primitive shapes (?@Jondolf, ?@NiseVoid)
- ECS
  - [ ] Dynamic queries (**claimed:** @nicopap)
  - [ ] Entity optimizations (?@Bluefinger, ?@notverymoe, @nicopap)
  - [ ] Split `WorldQuery` (?@wainwrightmark)
  - [ ] Auto-insert sync points (?@hymm, ?@JoJoJet)
- Rendering
  - [ ] Minimal reflection probes(?@pcwalton)
  - [ ] Exposure settings (?@coreh)
  - [ ] Light maps (?@pcwalton)
  - [ ] Sprite slicing (9-slices) (?@ManevilleF)
  - [ ] Light `RenerLayers` (?@robtfm)
  - [ ] Approximate indirect specular occlusion (?@aevyrie)
  - [ ] Unload render assets from RAM (?@JMS55, ?@mockersf)
  - [ ] Bind group layout entries (?@IceSentry, ?@JMS55)
- [ ] Camera-driven UI (?@bardt, @nicopap)
- [ ] Gizmo configs (**claimed:** @nicopap)
- [ ] GLTF extensions (?@CorneliusCornbread)
- [ ] Support all types of animation interpolation from GLTF (?@mockersf)

## `winit` and `wgpu` updates

<details><summary>winit and wgpu updates</summary>

### `winit` 0.29

Strongly involved: @Vrixyz, @mockersf, @daxpedda, @nicopap

* update PR: https://github.com/bevyengine/bevy/pull/10702
    * Followup fixes: https://github.com/bevyengine/bevy/pull/11227, https://github.com/bevyengine/bevy/pull/11245
* Logical key data: https://github.com/bevyengine/bevy/pull/11400

### `wgpu` 0.18 and 0.19

Strongly involved: @elabajaba

* 0.19: https://github.com/bevyengine/bevy/pull/11280
* 0.18: https://github.com/bevyengine/bevy/pull/10266

</details>

## Primitive shapes

Strongly involved: @Jondolf, @NiseVoid

* RFC: https://github.com/bevyengine/rfcs/pull/12
* Primitives: https://github.com/bevyengine/bevy/pull/10466
* Bounding volumes: https://github.com/bevyengine/bevy/pull/10946, https://github.com/bevyengine/bevy/pull/11336
* Bounding volume intersections: https://github.com/bevyengine/bevy/pull/11439

Many other smaller PRs: https://github.com/bevyengine/bevy/pull/10620, https://github.com/bevyengine/bevy/pull/10884, https://github.com/bevyengine/bevy/pull/11172, https://github.com/bevyengine/bevy/pull/11514, https://github.com/bevyengine/bevy/pull/11432

## ECS

<details><summary>ECS updates</summary>

### Dynamic queries

Strongly involved: @james-j-obrien

* https://github.com/bevyengine/bevy/pull/9774

### Entity representation optimizations

Strongly involved: @Bluefinger, @notverymoe

* Unified identifier: https://github.com/bevyengine/bevy/pull/9797
* Niche optimization: https://github.com/bevyengine/bevy/pull/9907
* Minor optimizations:
    * https://github.com/bevyengine/bevy/pull/10811 @bushrat011899
    * https://github.com/bevyengine/bevy/pull/10558 @Bluefinger
    * https://github.com/bevyengine/bevy/pull/10519 @scottmcm
    * https://github.com/bevyengine/bevy/pull/6773 @james7132

### Split `WorldQuery`

Strongly involved: @wainwrightmark, @taizu-jin

* https://github.com/bevyengine/bevy/pull/9918
* https://github.com/bevyengine/bevy/pull/10779

### Auto insert sync points

Strongly involved: @hymm, @JoJoJet

* https://github.com/bevyengine/bevy/pull/9822

</details>

## Rendering

<details><summary>Rendering updates</summary>

### Minimal reflection probes

Strongly involved: @pcwalton

* https://github.com/bevyengine/bevy/pull/11366

### Exposure settings

Strongly involved: @JMS55, @GitGhillie, @superdump, @coreh

* https://github.com/bevyengine/bevy/pull/11347

### Light maps

Strongly involved: @pcwalton

* https://github.com/bevyengine/bevy/pull/10231

### Sprite slicing and tiling (9-slice)

Strongly involved: @ManevilleF

* https://github.com/bevyengine/bevy/pull/5103
* https://github.com/bevyengine/bevy/pull/10588

#### Light `RenderLayers`

Strongly involved: @robtfm

* https://github.com/bevyengine/bevy/pull/10742

#### Approximate indirect specular occlusion

Strongly involved: @aevyrie

* https://github.com/bevyengine/bevy/pull/11152

#### Unload render assets from RAM

Strongly involved: @JMS55

* https://github.com/bevyengine/bevy/pull/10520

#### Bind group layout entries 

Strongly involved: @IceSentry

* https://github.com/bevyengine/bevy/pull/10224


</details>

## Camera-driven UI

Strongly involved: @oceantume, @bardt

* https://github.com/bevyengine/bevy/pull/10559

## Others


<details><summary>Other updates</summary>

### Gizmo configs

Strongly involved: @jeliag, @nicopap

* https://github.com/bevyengine/bevy/pull/10342
* https://github.com/bevyengine/bevy/pull/9187 

#### GLTF extensions

Strongly involved: @CorneliusCornbread

* https://github.com/bevyengine/bevy/pull/11138

#### Support all types of animation interpolation from GLTF

Strongly involved: @mockersf

* https://github.com/bevyengine/bevy/pull/10755

</details>

hymm commented:
I can take "auto sync points".

I think the list above is missing the texture atlas rework from alice's discord list.

ManevilleF commented:
I can take the texture atlas rework and texture slicing

wainwrightmark commented:
Happy to write up the Split WorldQuery section.

pcwalton commented:
I can write the lightmaps and reflection probes sections.

alice-i-cecile commented:
Awesome. I'm going to fill out some stub sections momentarily, then please make PRs to this branch :)

bardt commented:
I can write Camera-driven UI section

Jondolf commented:
Me and @NiseVoid can take the primitive shapes; I'll probably write up the beginning, explaining what they are and how they are useful, @NiseVoid will make something on bounding volumes and intersection tests, and I'll make a quick section for primitive meshing.

Bluefinger commented:
I could write up about the Entity optimisations, since a fair bit of the work was all fairly related/overlapping.

IceSentry commented:
I'll do the bind group layout entries

Trashtalk217 commented:
Since @Nathan-Fenner did the original implementation, I think they should get first dibs on writing the release notes for that. If they're busy or not interested, I could do that section.

Nathan-Fenner commented:
@Trashtalk217 Hi! I don't think I have the bandwidth to write anything right now, so please feel free to take the section (or let someone else do it)

alice-i-cecile commented:
@DasLixou (or @rodolphito), I've added a section for https://github.com/bevyengine/bevy/pull/10644

Similarly, @Trashtalk217, I've added a section for https://github.com/bevyengine/bevy/pull/10153

DasLixou commented:
Depends on what you expect to be written i might be up to it

hymm commented:
asset transformers might deserve a section https://github.com/bevyengine/bevy/pull/11260

bushrat011899 commented:
I could write the section on [extenion-less assets](https://github.com/bevyengine/bevy/pull/10153) if you'd like @Trashtalk217? Happy to defer if you guys would prefer though!

Trashtalk217 commented:
> I could write the section on [extenion-less assets](https://github.com/bevyengine/bevy/pull/10153) if you'd like @Trashtalk217? Happy to defer if you guys would prefer though!

Given that I had nothing to do with thatt PR and I'm not sure why I was tagged, I'd say: go right on ahead.

alice-i-cecile commented:
Sorry, I mistagged!

mockersf commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,199 @@
+To update an existing Bevy App or Plugin to **Bevy 0.13**, check out our [0.12 to 0.13 Migration Guide](/learn/migration-guides/0.12-0.13/).
+
+Since our last release a few months ago we've added a _ton_ of new features, bug fixes, and quality of life tweaks, but here are some of the highlights:
+
```
```suggestion

<!-- more -->

```

CorneliusCornbread commented:
I'm gonna see if I can set aside some time to write up the GLTF extension section in the next week or so. I don't think it should be too long tbh, considering it's pretty simple in concept.

coreh commented:
> Exposure settings (?@coreh)

I'm not _super_ confident on my ability to explain the changes made to exposure, my involvement was mostly on getting transmission and fog fixed to work with the new exposure code.

Ideally someone with a better grasp of the optics / units involved would take this? If no-one's available I can give it a try, but it probably won't be as good.

JMS55 commented:
I can write up Unload render assets from RAM, Async pipeline compilation (as well as the naga_oil 0.13 shader caching changes if @robtfm doesn't mind, feels like these two should be grouped), and Multithreaded command buffer encoding. The latter two haven't been merged yet but should be by the time we release 0.13.

Elabajaba commented:
I'll write up a small thing for the wgpu upgrades, and some general shader changes people will need to fix for custom shaders.

hymm commented:
Should add a section for system stepping.

hymm commented:
Need a section for https://github.com/bevyengine/bevy/pull/11660

Jondolf commented:
bevyengine/bevy#10878 is a basic but very user-facing breaking change. Calling `.into()` when adding assets has probably been very common in projects, and people now need to remove it. Might be worth a short section? Or just leave it to the migration guide only

dmlary commented:
I'll write up system stepping.

hymm commented:
**Potential additions:**
* Async Pipeline Compilation
* Batching Improvements

**Interesting, but not sure if they deserve a full section**
* Unified identifer for entities & relations (https://github.com/bevyengine/bevy/pull/9797)
* Change Entity::generation from u32 to NonZeroU32 for niche optimization (https://github.com/bevyengine/bevy/pull/9907)

Should we have a section for Added, Changed, Deprecated API's highlights section like the rust release blogs? Probably would need someone besides Alice to go through all the PR's a compile a list.
* Add run conditions for executing a system after a delay (https://github.com/bevyengine/bevy/pull/11095)
* Deprecated Various Component Methods from Query and QueryState #9920
* get_resource_ref

Bluefinger commented:
> **Interesting, but not sure if they deserve a full section**
> 
>     * Unified identifer for entities & relations ([Unified identifer for entities & relations bevy#9797](https://github.com/bevyengine/bevy/pull/9797))
> 
>     * Change Entity::generation from u32 to NonZeroU32 for niche optimization ([Change Entity::generation from u32 to NonZeroU32 for niche optimization bevy#9907](https://github.com/bevyengine/bevy/pull/9907))

That would be a shame to remove these, as https://github.com/bevyengine/bevy/pull/9907 paved the way for https://github.com/bevyengine/bevy/pull/10558 to be considered as an improvement, while also not preventing https://github.com/bevyengine/bevy/pull/9797 from being doable with the approach taken. These are very much related and overlap each other considerably, and when these all merged, it was a very...

![27d](https://github.com/bevyengine/bevy-website/assets/3116268/b6091f7e-68ed-4c54-aa9d-91113b61c5b1)
 
...moment. Maybe it can be shortened a bit for the first part of that section? I still think it would be a shame to lose some of the context of _why_ https://github.com/bevyengine/bevy/pull/10558 was able to be merged when previous attempts were not successful.

hymm commented:
oh, my bad. I didn't realize they were already include as part of entity optimizations.

Bluefinger commented:
> oh, my bad. I didn't realize they were already include as part of entity optimizations. I was proposing them for inclusiong

Ahh, no worries. I misread what you were trying to say in that case.

bardt commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,920 @@
+
+TODO.
+
+## Type-Safe Labels for the `RenderGraph`
```
There might be a mistake made while merging. This title should be `## Camera-Driven UI`; and where should be this title, I don't know.

superdump commented:
I think the phase sorting changes by @Elabajaba  should have a release entry. They bring a good perf bump for scenes which lots of opaque geometry which could be batched but weren’t in 0.12 due to sorting based on depth where they are now sorted by pipeline id and then mesh asset id: <https://github.com/bevyengine/bevy/pull/11671>

Bluefinger commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1260 @@
+
+### Changing `Entity`'s Layout
+
+This is a story told in two parts, with part one dealing with two PRs: The "Unified Identifier for entities & relations" PR ([#9797](https://github.com/bevyengine/bevy/pull/9797) by @Bluefinger) and "Change Entity::generation from u32 to NonZeroU32 for niche optimization" ([#9907](https://github.com/bevyengine/bevy/pull/9907) by @notverymoe). Fundamentally, they both involved changing the layout  ...
```
```suggestion
This is a story told in two parts, with part one dealing with two PRs: The "Unified Identifier for entities & relations" PR ([#9797](https://github.com/bevyengine/bevy/pull/9797) by @Bluefinger) and "Change Entity::generation from u32 to NonZeroU32 for niche optimization" ([#9907](https://github.com/bevyengine/bevy/pull/9907) by @notverymoe). Fundamentally, they both involved changing the layout and assumptions of `Entity`, unlocking building blocks for relations and improving memory/codegen for `Entity`.
```
Singular/Plural mix here, probably better to remove the "a". Maybe even use "some" instead?

james7132 commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1286 @@
+
+<div class="release-feature-authors">authors: @Elabajaba</div>
+
+In Bevy 0.13 we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to do things like compiling shaders asynchronously to avoid shader compilation stutters, and to multithread draw call creation for better performance.
```
This should probably link https://github.com/bevyengine/bevy/pull/9172 in some way to show how we've taken advantage of this.

Elabajaba replied:
Yeah, I wrote this up this way when @JMS55 was going to do a writeup for both async pipeline compilation and multithreaded render command encoding.

james-j-obrien commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1286 @@
+`Query<&Transform>` you have a `Query<(&Transform, &Velocity), With<Enemy>>`?
+Well now you can by using the `Query::transmute_lens` method. Query transmutes
+allow you to change a query into different query types as long as the
+componenets accessed are a subset of the original query. If you do try to access
```
```suggestion
components accessed are a subset of the original query. If you do try to access
```

aevyrie commented:
I'll take specular occlusion.

mahulst commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1311 @@
+
+This change unlocks a variety of new UI experiences, including split-screen multiplayer, UI in multiple windows, displaying non-interactive UI in a 3D world, and more.
+
+![Split-screen with independent UI roots](split-screen.png)
```
These split screen UIs look almost exactly the same. Should the screenshot not express more clearly you can now have a different UI per camera?

inodentry commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1518 @@
+inside the original query. It's not a expensive operation, but you should
+probably avoid doing it inside a hot loop.
+
+## Entity Optimizations
```
This section is extremely technical and goes into way more detail than feels appropriate for a release blog post. It's also way too early in the blog. IMO the sections should be ordered as to highlight the most exciting user-facing high-level features first, and the most technical low-level improvements to engine internals last.

I suggest to either simplify / shorten it, omitting the most low-level details.

Also please move it way further down towards the end of the blog.

Bluefinger replied:
First off, I want to add my agreement that this probably should go more towards the end of the blog, for the exact reasons above: User facing stuff first, technical/perf improvements last.

Simplifying this will probably entail removing the details/context as to _why_ such a change was important (but also why it didn't land previously). This sort of exploration is I think beneficial to the rust community at large that could benefit from the visibility of the techniques used/merged into bevy. *That said*, maybe this is a candidate for a follow-up post? A post where we can indeed go into more depth about these topics. Summarise for the release, expand in the follow-up.

Likewise, we could take the opportunity to write more of that sort of content to the bevy website. There's no shortage of stuff we could talk about in terms of highly technical challenges Bevy faces in trying to optimise, improve and further the state of art for ECS in Rust and in the game engine space. Also a nice way to keep some visibility/activity for the Bevy project outside of Discord instead of every 3/4 months (apart from the occasional other post).

alice-i-cecile replied:
Yeah, I really like this content, but I've been chewing on how to reduce its length. I'm personally in favor of focusing on impact, and moving the crunchy details into either an official blog post, or a third-party blog post that we boost.

alice-i-cecile replied:
This has been greatly reduced and moved to the end <3 Super cool stuff, but distracting.

Bluefinger commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1676 @@
+
+The assembly output was compared as well between what was on main branch versus the PR, with no tangible differences being seen between the old `Query::for_each` and the new `QueryIter::for_each()` output, validating the approach and ensuring the internal iteration optimizations were being applied.
+
+As a plus, the same iternal iteration optimizations in `Query::par_for_each` now reuse code from `for_each`, deduplicating code there as well and enabling users to make use of `par_iter().for_each()`. As a whole, this means there's no longer any need for `Query::for_each`, `Query::for_each_mut`, `Query::_par_for_each`, `Query::par_for_each_mut` so these methods have been deprecated for 0.13 and w ...
```
```suggestion
As a plus, the same internal iteration optimizations in `Query::par_for_each` now reuse code from `for_each`, deduplicating code there as well and enabling users to make use of `par_iter().for_each()`. As a whole, this means there's no longer any need for `Query::for_each`, `Query::for_each_mut`, `Query::_par_for_each`, `Query::par_for_each_mut` so these methods have been deprecated for 0.13 and will be removed in 0.14.
```
Oops, found a typo in my work after re-rereading :smile:

Bluefinger commented on content/news/2024-02-03-bevy-0.13/index.md:
```diff
@@ -0,0 +1,1676 @@
+* [#9797]: created a unified identifier type, paving the path for us to use the same fast, complex code in both our `Entity` type and the much-awaited relations
+* [#9907]: allowed us to store `Option<Entity>` in the same number of bits as `Entity`, by changing the layout of our Entity type to reserve exactly one `u64` value for the `None` variant
+* [#10519]: swapped us to a manually crafted `PartialEq` and `Hash` implementation for `Entity` to improve speed and save instructions in our hot loops
+* [#10558]: took the same approach, and further optimized our `PartialOrd` and `Ord` implementations!
```
```suggestion
* [#10558]: combined the approach of [#9907] and [#10519] to optimize `Entity`'s layout further , and optimized our `PartialOrd` and `Ord` implementations!
```
This is a nit, but I think it better describes/summarises the follow-up and what the work was built upon.

xyzw-io commented:
Materials from Jarl game for splash, per our conversation in Discord. Tried to include a screenshot, short and long video and gifs. Feel free to use any of these and/or edit them if you need: https://drive.proton.me/urls/8N6BKM603R#Ipsw82zEed52

Here are the links for this project (if only one can be used, the first one is the best):

1. [https://jarl-game.com](https://jarl-game.com)
2. [https://discord.gg/jarl](https://discord.gg/jarl)
3. [https://youtube.com/@Jarl-Game-com](https://youtube.com/@Jarl-Game-com)
4. [https://reddit.com/r/jarlgame/](https://reddit.com/r/jarlgame/)
5. [https://twitter.com/jarl_game](https://twitter.com/jarl_game)

TrialDragon commented:
Should the splash image GIF be converted into a format that is smaller since folks will see it loading on opening the page? Atm it's 11.6 MB; as a WebP it could be approx. 5.4 MB.

cart commented:
> Should the splash image GIF be converted into a format that is smaller since folks will see it loading on opening the page? Atm it's 11.6 MB; as a WebP it could be approx. 5.4 MB.

Good call. Done!

Bluefinger commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3208 @@
+[#2372]: https://github.com/bevyengine/bevy/pull/2372
+[#3788]: https://github.com/bevyengine/bevy/pull/3788
+
+## Porting `Query::for_each` to `QueryIter::fold` override
```
This _was_ a sub-heading at one point, but was elevated, so maybe this needs the `<div class="release-feature-author">` part here as well.

Bluefinger commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3208 @@
+
+As a plus, the same internal iteration optimizations in `Query::par_for_each` now reuse code from `for_each`, deduplicating code there as well and enabling users to make use of `par_iter().for_each()`. As a whole, this means there's no longer any need for `Query::for_each`, `Query::for_each_mut`, `Query::_par_for_each`, `Query::par_for_each_mut` so these methods have been deprecated for 0.13 and  ...
+
+## Reducing `TableRow` `as` Casting
```
Same as above with the author section

Bluefinger commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3208 @@
+But the benefit was a less error-prone API and more robust code.
+With a complex unsafe codebase like `bevy_ecs`, every little bit helps.
+
+## Events Live Longer
```
This is also missing an author's section.

xyzw-io commented:
@cart here is the same webp, but made from original video with higher res better preserving pixel art details. I hope 3.8mb is acceptable size): [output.webp.zip](https://github.com/bevyengine/bevy-website/files/14319263/output.webp.zip)

Attached as zip, because GH does not allow webp attachments.

NiklasEi commented:
@xyzw-io the game looks awesome! Would it possible to get a clip without the cursor in the middle of the scene?

JMS55 commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3215 @@
+* **Irradiance Volumes / Voxel Global Illumination:** A baked form of global illumination that samples light at the centers of voxels within a cuboid (baked externally in programs like Blender).
+* **Approximate Indirect Specular Occlusion**: Improved lighting realism by reducing specular light leaking via specular occlusion.
+* **Reflection Probes**: A baked form of axis aligned environment map that allows for realistic reflections for static geometry (baked externally in programs like Blender)
+* **Primitive shapes:** basic shapes are a core building block of both game engines and video games: we've added a polished, ready-to-use collection of them!
```
Some of these bullet points do not start with a capitalized word

JMS55 commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3215 @@
+
+<div class="release-feature-authors">authors: @Elabajaba, @JMS55</div>
+
+In **Bevy 0.13** we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to [compile shaders asynchronously](https://github.com/bevyengine/bevy/pull/10812) to avoid shader compilation stutters and [multithread draw call creation](https://github.com/bevyengine/bevy/pull/9172 ) for bette ...
```
```suggestion
In **Bevy 0.13** we upgraded from `wgpu` 0.17 to `wgpu` 0.19, which includes the long awaited `wgpu` [arcanization](https://gfx-rs.github.io/2023/11/24/arcanization.html) that allows us to [compile shaders asynchronously](https://github.com/bevyengine/bevy/pull/10812) to avoid shader compilation stutters and [multithread draw call creation](https://github.com/bevyengine/bevy/pull/9172 ) for better performance in CPU-bound scenes.
```

JMS55 commented on content/news/2024-02-17-bevy-0.13/index.md:
```diff
@@ -0,0 +1,3215 @@
+in many games, memory usage is the biggest limitation on the resolution and polygon count of the game!
+Moreover, actually transferring that data from system RAM (used by the CPU) to the VRAM (used by the GPU) can be a real performance bottleneck.
+
+**Bevy 0.13** adds the ability to unload this data from system RAM, once it has been succesfully transferred to VRAM. However, unloading the data from the CPU [can result in bugs], and make it harder to actually inspect or modify that data from your other game systems.
```
[can result in bugs]

Broken link? Also, this section is kind of unclear. Result in bugs is very vague. The actual problem is users or libraries like rapier depending on mesh data in a system in order to generate colliders and stuff at runtime, which need the asset data in memory until the collider is generate, and then it can be unloaded. 

Also this section could be a bit more clear, the reason you'd want to unload render assets is to save on RAM since you only need it in VRAM for the renderer, and we strongly recommend users to set the persistence policy to render only.

alice-i-cecile replied:
This was intended to link to the revert PR.

cart replied:
Adressing this now

cart replied:
Link isn't actually broken. It just renders as broken in vscode's syntax highlighting
